internal/monitors/elasticsearch/client/nodestatsfetcher.go

```diff
@@ -0,0 +1,164 @@
+package client
+
+import (
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+)
+
+const (
+	transportStatsGroup = "transport"
+	httpStatsGroup      = "http"
+	//indicesStatsGroup = "indices"
+	jvmStatsGroup        = "jvm"
+	threadpoolStatsGroup = "thread_pool"
+	processStatsGroup    = "process"
+)
+
+// Method to fetch node stats
+func GetNodeStatsDatapoints(nodeStatsOutput *NodeStatsOutput, defaultDims map[string]string, selectedThreadPools []string, nodeStatsGroups []string) []*datapoint.Datapoint {
+	var out []*datapoint.Datapoint
+	for _, nodeStats := range nodeStatsOutput.NodeStats {
+		out = append(out, GetNodeStatsDatapointsHelper(nodeStats, defaultDims, selectedThreadPools, nodeStatsGroups)...)
+	}
+	return out
+}
+
+func GetNodeStatsDatapointsHelper(nodeStats NodeStats, defaultDims map[string]string, selectedThreadPools []string, nodeStatsGroups []string) []*datapoint.Datapoint {
+	var dps []*datapoint.Datapoint
+	dps = append(dps, nodeStats.Jvm.fetchJVMStats(utils.StringInSlice(jvmStatsGroup, nodeStatsGroups), defaultDims)...)
```

Comment:
Convert `nodeStatsGroups` to a `map[string]bool`, which is like a set.  That way all you have to do is pass `nodeStatsGroupsSet[jvmStatsGroup]` which is O(1) instead of O(n) to the number of groups.

Reply:
Using `bool` options now. One per group, as mentioned in an earlier comment.